jaxquantum/devices/superconducting/__init__.py

```diff
@@ -7,6 +7,7 @@
 from .kno import *  # noqa
 from .transmon import *  # noqa
 from .tunable_transmon import *  # noqa
+from .truncated_transmon import *  # noqa
```

Comment:
What is the purpose of this truncated_transmon class? Is it not covered by the transmon class? If so, let's delete it in this PR.

Reply:
I think it might have been an older class, and yeah all relevant functionality should be covered by the usual transmon. I have deleted it.

The only thing the truncated transmon does differently is that it constructs the full cosine potential in the Fock basis with a nonlinearity of cos(phi) - EJ * phi^2 /2, rather than truncating that nonlinearity at 6th order. 

However, that's mostly not valid anyways in the Fock basis so fine to remove.